### ISSUE TITLE
Flow strict mode

### ISSUE THREAD

**Issue by hendrikniemann** [gajus/slonik#12]
I am currently working on a new project that uses [Flow's strict mode](https://flow.org/en/docs/strict/) pretty much everywhere. If I use slonik from these files it warns about slonik not being in strict mode (this can be fixed by using _strict-local_ instead). I wanted to suggest to switch to strict mode for slonik. Projects like `graphql-js` are already in strict mode. If so, I am also glad to try to do this change and send a PR.

Thanks for the awesome work with slonik and postloader!

**Comment by gajus** [comment]
Go ahead with the PR.

Is it just a single change in the `.flowconfig` or does the change need to be done in every file?

**Comment by hendrikniemann** [comment]
So I looked into this a bit and I think I didn't quite understand the way _flow strict_ works before. The main benefit of _flow strict_ is that it turns linting rules into errors and the strict behaviour can be opted in for each file individually (by adding `// @flow strict`). Which rules strict applies can be configured in the `.flowconfig`.

On `init` flow creates some default rules for strict which contains `nonstrict-import`, which is a nice rule to ensure maximum type safety but turns out not to be very practical. This rule requires `strict` mode modules to only require other strict mode modules - the reason I opened this issue, so that slonik would also be in strict mode. But just turning it into strict mode is not enough because flow will not apply the dependencies' `.flowconfig` strict rules to the dependencies' code but the project's.

In short the `nonstrict-import` rule only works if *all* (also nested) dependencies are in strict mode and apply the rule. For most projects in the JS OSS ecosystem (e.g. slonik with 11 dependencies) this is just unrealistic. I don't think there would be much benefit from moving to strict mode for slonik.

**Comment by gajus** [comment]
Well, I am happy to do my share and move Slonik to strict mode. You can leave it to me. Will do it sometime in the future.
